packages/main/src/Calendar.ts

```diff
@@ -304,6 +331,47 @@ class Calendar extends CalendarPart {
 		});
 	}
 
+	_isValidCalendarDate(dateString: string): boolean {
+		const date = this.getFormat().parse(dateString);
+		return !!date;
+	}
+
+	_populateSpecialCalendarDates(selectedItemType?: string): void {
```

Comment:
Can we switch this one to getter and to pass it as it is instead of storing every preprocessor to the property? maybe with 1 or 2 private fields we can achieve the same effect